Makefile

```diff
@@ -33,23 +39,46 @@ pre-pr: tidy generate lint test
 clean:
 	-@rm test/testdata/x.wasm
 
-sdk_srcs := $(wildcard ./basm/**/*.go)
-wasm_src_dir := $(shell realpath ./test/testdata)
-wasm_src := $(wasm_src_dir)/main.go
-wasm_out := $(wasm_src_dir)/x.wasm
+sdk_src := $(wildcard ./basm/**/*.go)
+wasm_src := $(testdata_dir)/main.go
+wasm_out := $(testdata_dir)/x.wasm
 
 .PHONY: wasm
 wasm: $(wasm_out)
 
-$(wasm_out): $(wasm_src) $(sdk_srcs)
+$(wasm_out): $(wasm_src) $(sdk_src)
 	@echo "Building WASM module..."
 	@tinygo build -o $@ -target=wasi $<
 
-test_scripts := $(shell realpath $(wildcard ./test/scripts/*.txtar))
+# Set default values for BASM_ variables. These can be overridden by the user
+# e.g. `BASM_USER_SECRET=mysecret make test`
+BASM_PLATFORM ?= plain
+BASM_CODE_MEASURE ?= plain
+BASM_AUTH_TOKEN ?= auth token
+BASM_HOST ?= local-server
+BASM_USER_SECRET ?= this is a bearer token
+# Export all BASM_ variables for use in recipes
+export $(filter BASM_%,$(.VARIABLES))
```

Comment:
experimenting with another way to pass env vars to make tasks here. I found the `make VAR=value VAR2=value test-live` pattern in delphi to be a bit strange, so here we allow the makefile to specify some defaults values, but otherwise it will pull these values from the environment automatically.